src/meta/src/rpc/server.rs

```diff
@@ -171,316 +153,312 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
+                    worker_node_parallelism: 5,
+                })
+                .await;
+            let is_leader = resp.is_ok();
+
+            if is_leader {
+                leader_count += 1;
+            }
+        }
+        leader_count
     }
 
-    let catalog_manager = Arc::new(CatalogManager::new(env.clone()).await.unwrap());
+    // Writing these tests as separate functions instead of one loop, because functions get executed
+    // in parallel
+    #[tokio::test]
+    async fn test_single_leader_setup_1() {
+        let v = _setup_n_nodes(1, 1234).await;
+        let leader_count = _number_of_leaders(1, 1234, 5678).await;
+        assert_eq!(
+            leader_count, 1,
+            "Expected to have 1 leader, instead got {} leaders",
+            leader_count
+        );
+        for element in v {
+            element.0.abort();
+        }
+    }
 
-    let (barrier_scheduler, scheduled_barriers) =
-        BarrierScheduler::new_pair(hummock_manager.clone(), env.opts.checkpoint_frequency);
+    #[tokio::test]
+    async fn test_single_leader_setup_3() {
+        let v = _setup_n_nodes(3, 2345).await;
+        let leader_count = _number_of_leaders(3, 2345, 6789).await;
+        assert_eq!(
+            leader_count, 1,
+            "Expected to have 1 leader, instead got {} leaders",
+            leader_count
+        );
+        for element in v {
+            element.0.abort();
+        }
+    }
 
-    let source_manager = Arc::new(
-        SourceManager::new(
-            barrier_scheduler.clone(),
-            catalog_manager.clone(),
-            fragment_manager.clone(),
-        )
-        .await
-        .unwrap(),
-    );
-
-    let barrier_manager = Arc::new(GlobalBarrierManager::new(
-        scheduled_barriers,
-        env.clone(),
-        cluster_manager.clone(),
-        catalog_manager.clone(),
-        fragment_manager.clone(),
-        hummock_manager.clone(),
-        source_manager.clone(),
-        meta_metrics.clone(),
-    ));
-
-    {
-        let source_manager = source_manager.clone();
-        tokio::spawn(async move {
-            source_manager.run().await.unwrap();
-        });
+    #[tokio::test]
+    async fn test_single_leader_setup_10() {
+        let v = _setup_n_nodes(10, 3456).await;
+        let leader_count = _number_of_leaders(10, 3456, 7890).await;
+        assert_eq!(
+            leader_count, 1,
+            "Expected to have 1 leader, instead got {} leaders",
+            leader_count
+        );
+        for element in v {
+            element.0.abort();
+        }
     }
 
-    let stream_manager = Arc::new(
-        GlobalStreamManager::new(
-            env.clone(),
-            fragment_manager.clone(),
-            barrier_scheduler.clone(),
-            cluster_manager.clone(),
-            source_manager.clone(),
-            hummock_manager.clone(),
-        )
-        .unwrap(),
-    );
+    #[tokio::test]
+    async fn test_single_leader_setup_100() {
```

Comment:
I think only testing under one or some like 3 or 5 meta nodes is already enough.

Reply:
Alright. I will go for 1, 3 and 5 then :)